src/keystore/InMemoryKeystore.ts

```diff
@@ -294,12 +299,21 @@ export default class InMemoryKeystore implements Keystore {
           )
         }
 
+        const keyMaterial = getKeyMaterial(topicData.invitation)
+        const ciphertext = await encryptV2(payload, keyMaterial, headerBytes)
+        const thirtyDayPeriodsSinceEpoch = Math.floor(
+          Date.now() / 1000 / 60 / 60 / 24 / 30
+        )
+        const info = `${thirtyDayPeriodsSinceEpoch}-${this.accountAddress}`
+        const hmac = await generateHmacSignature(
+          keyMaterial,
+          new TextEncoder().encode(info),
+          headerBytes
```

Comment:
```suggestion
          ciphertext
```
I believe this is right

Reply:
So, the ciphertext in this case is an object with a payload, nonces, and a salt. It doesn't get serialized until the whole message gets serialized. 

Could create the hmac on just the payload, or define an order to concat them, but the header bytes seemed the easiest because they're already a byte array. We do have other mitigations to prevent replays of headers.